packages/ia-pic-uploader/src/ia-pic-uploader.ts

```diff
@@ -0,0 +1,834 @@
+          console.log('task(s) done!');
+          clearInterval(metadataApiInterval);
+          this.taskStatus = 'reloading page with your image';
+          window.location.reload();
+        }
+      });
+    }, 2000);
+  }
+
+  /**
+   * fuction to handel closing functionalities
+   */
+  cancelFile() {
+    const imagePreview = this.selfSubmitEle?.querySelector('.image-preview');
+
+    //  clear file input
+    if (this.fileSelector) this.fileSelector.value = '';
+
+    this.showDropper = false;
+    this.showLoadingIndicator = false;
+    this.fileValidationError = '';
+
+    this.selfSubmitEle?.classList.add('hidden');
+    this.profileSection?.classList.remove('hover-class');
+
+    while (
+      imagePreview?.firstChild &&
+      imagePreview.removeChild(imagePreview.firstChild)
+    );
+  }
+
+  /**
+   * function to render loader indicator
+   * @returns {HTMLElement} | <ia-activity-indicator>
+   */
+  get loadingIndicatorTemplate() {
+    return html` <ia-activity-indicator
+      mode="processing"
+      class="go-button-loading-icon"
+    ></ia-activity-indicator>`;
+  }
+
+  get plusIconTemplate() {
+    return html`<div
+      class="plusIcon ${this.showLoadingIndicator ? 'pointer-none' : ''}"
+      @keyup=""
+      @click=${() => {
+        this.dropRegion?.click();
+      }}
+    >
+      <span>&#43;</span>
+    </div>`;
+  }
+
+  /**
+   * function to render self submit form template
+   * @returns {HTMLElement}
+   */
+  get selfSubmitFormTemplate() {
+    const formAction = encodeURIComponent(
+      `${this.endpoint}?identifier=${this.identifier}&submit=1`
+    );
+
+    return html`
+      <div
+        class="self-submit-form hidden
+      "
+      >
+        <button
+          class="close-button ia-button 
+          ${(!this.showDropper && this.fileValidationError === '') ||
+          this.showLoadingIndicator
+            ? 'hidden'
+            : ''}
+          ${this.showLoadingIndicator ? 'pointer-none' : ''}"
+          @click=${() => {
+            this.cancelFile();
+          }}
+        >
+          &#10060;
+        </button>
+        ${this.showLoadingIndicator
+          ? this.loadingIndicatorTemplate
+          : this.plusIconTemplate}
+        <span
+          class="drag-text ${this.showLoadingIndicator ? 'pointer-none' : ''}"
+          @keyup=""
+          @click=${() => {
+            this.dropRegion?.click();
+          }}
+          >${this.taskStatus
+            ? this.taskStatus
+            : 'Drag & Drop an image file here or'}</span
+        >
+        <button
+          id="file-picker"
+          @click=${() => {
+            this.dropRegion?.click();
+          }}
+          class="ia-button primary ${this.showLoadingIndicator
+            ? 'pointer-none hidden'
+            : ''}"
+        >
+          Pick image to upload
+        </button>
+        <div
+          class="validationErrorDiv"
+          style="display:${this.fileValidationError === '' ? 'none' : 'block'}"
```

Comment:
Q: Can we use a CSS class for this instead of inline styles?

Reply:
i will use class instead of inline.
Thanks